api/v1alpha1/externaldns_types.go

```diff
@@ -501,6 +497,13 @@ type ExternalDNSServiceSourceOptions struct {
 	ServiceType []corev1.ServiceType `json:"serviceType,omitempty"`
 }
 
+type ExternalDNSOpenShiftRouteOptions struct {
+	// If source is openshift-route then you can pass the ingress controller name. Based on this name external-dns will select the respective router.
+	// +kubebuilder:validation:Required
+	// +required
+	RouterName string `json:"routerName"`
```

Comment:
How do you know the router name from a route instance?

Reply:
It's in the route's status: [route api](https://github.com/openshift/api/blob/be1be0e89115702f8b508d351c4f5c9a16e5ae95/route/v1/types.go#L166).